bubble_sort.s

```diff
@@ -0,0 +1,234 @@
+        lw      $a0, 0($t0)             # load data from address in $t0 into $a0
+        li	$v0, 4                  # code of print_string
+        syscall  
+
+	# scanf num elements of array
+	li 	$t0 , 0			# t0 = 0 - counter
+	lw 	$t3 , num		# t3 = num#	
+	scanarr:			# beginig of for
+		li	$v0 , 5			# code if read_int
+		syscall				# call scanf for int
+		sw 	$v0 , ($s0) 		# store int to the t1
+		add 	$s0 , $s0 , 4		# t1 += sizeof(int)
+		add 	$t0 , $t0 , 1		# t0++
+	blt 	$t0 , $t3 , scanarr	# condtiton of for
+	b	end_of_is_static
+# END OF scanf AND sbrk BLOCK
+
+	
+# STATIC ARRAY BLOCK
+	
+	static:
+
+	# saving address of array
+	la	$s0 , arr		# load array address in s0
+	la	$s1 , arr		# load array address in s1
+	
+	# Printing input array
+	li	$v0 , 4			# code of print string
+	la	$t0 , printarr		# load address of printarr to t0
+	lw	$a0 , ($t0)		# load data from t0
+	syscall
+
+	li      $t0 , 0                 # t0 = 0 - counter
+        lw      $t3 , num               # t3 = num     
+        showinputarr:                       # beginig of for
+                li      $v0 , 1                 # code if print_int
+		lw	$a0 , ($s0)		# a0 = *s0
+                syscall                         # call print for int
+                add     $s0 , $s0 , 4           # s0 += sizeof(int)
+                add     $t0 , $t0 , 1           # t0++
+		#print space
+		li	$v0 , 4			# code of print string
+		la	$t7 , space		# load address of spase if t6
+		lw	$a0 , ($t7)		# load data from address in t6
+		syscall				# call printf space
+        blt     $t0 , $t3 , showinputarr    # condtiton of for
+# END OF STATIC ARRAY BLOCK
+
+	end_of_is_static:
+	# now we have number of elements in num and address of array in s1
+	# calling bubble_sort
+	addi	$sp , $sp , -4		# move stack pointer on 1 word
+	sw	$ra , ($sp) 		# saving ra
+	lw	$a1 , num		# a1 = num
+	move 	$a0 , $s1		# a0 = address of array
+	jal bubble_sort
+
+	# printing array after sort
+	# printing main resbody
+        la      $t0, resbody            # load address of resbody into $t0
+        lw      $a0, 0($t0)             # load data from address in $t0 to $a0
+        li      $v0, 4                  # code of print_string
+        syscall
+
+	#print res
+	li      $t0 , 0                 # t0 = 1 - counter
+        lw      $t3 , num               # t3 = num
+        printsortarr:                        # beginig of for
+                li      $v0 , 4                 # code of print_string
+		la	$t7 , space		# load address of res in $t7
+		lw	$a0 , ($t7)		# load data from address in $t7 to $a0
+		syscall				# print " "
+		li	$v0 , 1
+		lw	$a0 , ($s1)		# a1 = *t1 - argument of print
+                syscall                         # call print for int
+                add     $s1 , $s1 , 4           # t1 += sizeof(int)
+                add     $t0 , $t0 , 1           # t0++
+        blt     $t0 , $t3 , printsortarr     # condtiton of for
+
+	# exit
+	addi	$sp , $sp , 4		# sp += sizeof(int)
```

Comment:
Looks like a bug: you increment the pointer first, and then read the value. IIRC, order should be opposite to push order.